test/e2e/chaos_test.go

```diff
@@ -240,6 +257,136 @@ func testKillAllMembers(parentCtx context.Context, client crclient.Client, clust
+			want := int32(1)
+			got := job.Status.Active
+			return want != 0 && want == got, fmt.Sprintf("wanted status active to be %d , got %d", want, got), nil
+		}}, e2eutil.WithInterval(5*time.Second), e2eutil.WithTimeout(10*time.Minute))
+
+		// The etcd cluster should eventually roll out completely
+		e2eutil.EventuallyObject(t, ctx, "etcd StatefulSet replicas to converge", func(ctx context.Context) (*appsv1.StatefulSet, error) {
+			sts := cpomanifests.EtcdStatefulSet(guestNamespace)
+			err := client.Get(ctx, crclient.ObjectKeyFromObject(sts), sts)
+			return sts, err
+		}, []e2eutil.Predicate[*appsv1.StatefulSet]{func(sts *appsv1.StatefulSet) (done bool, reasons string, err error) {
+			want := ptr.Deref(etcdSts.Spec.Replicas, 3)
+			got := sts.Status.ReadyReplicas
+			return want != 3 && want == got, fmt.Sprintf("wanted %d replicas in spec, got %d in status", want, got), nil
+		}}, e2eutil.WithInterval(5*time.Second), e2eutil.WithTimeout(30*time.Minute))
+	}
+}
+
+// testEtcdMemberMissing ensures that the etcd cluster can recover from a missing member
+func testEtcdMemberMissing(parentCtx context.Context, client crclient.Client, cluster *hyperv1.HostedCluster) func(t *testing.T) {
+	return func(t *testing.T) {
+		g := NewWithT(t)
+		ctx, cancel := context.WithCancel(parentCtx)
+		defer cancel()
+
+		guestNamespace := manifests.HostedControlPlaneNamespace(cluster.Namespace, cluster.Name)
+		t.Logf("Hosted control plane namespace is %s", guestNamespace)
+
+		// Find etcd pods in the control plane namespace
+		etcdSts := cpomanifests.EtcdStatefulSet(guestNamespace)
+		err := client.Get(ctx, crclient.ObjectKeyFromObject(etcdSts), etcdSts)
+		g.Expect(err).NotTo(HaveOccurred(), "failed to get etcd statefulset")
+
+		etcdPods := &corev1.PodList{}
+		err = client.List(ctx, etcdPods, &crclient.ListOptions{
+			Namespace:     manifests.HostedControlPlaneNamespace(cluster.Namespace, cluster.Name),
+			LabelSelector: labels.Set(etcdSts.Spec.Selector.MatchLabels).AsSelector(),
+		})
+		g.Expect(err).NotTo(HaveOccurred(), "failed to get etcd pods")
+
+		pod := randomPods(etcdPods.Items, 1)[0]
+		ep := fmt.Sprintf("https://etcd-client.%s.svc:2379", guestNamespace)
+		baseCommand := []string{
+			"/usr/bin/etcdctl",
+			"--cacert=/etc/etcd/tls/etcd-ca/ca.crt",
+			"--cert=/etc/etcd/tls/server/server.crt",
+			"--key /etc/etcd/tls/server/server.key",
+			fmt.Sprintf("--endpoints=%s", ep),
+		}
+
+		// Get ETCD member ID
+		innerCommand := fmt.Sprintf("member list | grep %s | awk '{print $1}' | tr -d ,", pod.Name)
+		memberDiscoveryCommand := append(baseCommand, innerCommand)
+
+		// Final etcd commands
+		command := append(baseCommand, "member", "remove", fmt.Sprintf("$(%s)", memberDiscoveryCommand))
+
+		t.Logf("Deleting ETCD Member: %s", pod.Name)
+		_, err = e2eutil.RunCommandInPod(ctx, client, pod.Spec.Containers[0].Name, pod.Namespace, command, pod.Spec.Containers[0].Name, 5*time.Minute)
```

Comment:
```suggestion
		_, err = e2eutil.RunCommandInPod(ctx, client, "etcd", pod.Namespace, command, pod.Spec.Containers[0].Name, 5*time.Minute)
```
this arg is for the component name, not container name.

Reply:
In this case is the same but as you mentioned, makes sense to hardcoded it.